zone/special_attacks.cpp

```diff
@@ -2127,105 +2127,109 @@ void Client::DoClassAttacks(Mob *ca_target, uint16 skill, bool IsRiposte)
 
 	int level_difference = GetLevel() - who->GetLevel();
 	bool success = false;
 
 	// Support for how taunt worked pre 2000 on LIVE - Can not taunt NPC over your level.
-	if ((RuleB(Combat, TauntOverLevel) == false) && (level_difference < 0) ||
-	    who->GetSpecialAbility(IMMUNE_TAUNT)) {
+	if (
+		!RuleB(Combat, TauntOverLevel) && 
+		level_difference < 0 ||
+	    who->GetSpecialAbility(IMMUNE_TAUNT)
+	) {
 		MessageString(Chat::SpellFailure, FAILED_TAUNT);
 		return;
 	}
 
 	// All values used based on live parses after taunt was updated in 2006.
-	if ((hate_top && hate_top->GetHPRatio() >= 20) || hate_top == nullptr || chance_bonus) {
+	if (
+		(hate_top && hate_top->GetHPRatio() >= 20) ||
+		!hate_top ||
+		chance_bonus
+	) {
 		// SE_Taunt this is flat chance
 		if (chance_bonus) {
 			success = zone->random.Roll(chance_bonus);
 		} else {
-			float tauntchance = 50.0f;
-
-			if (always_succeed)
-				tauntchance = 101.0f;
-
-			else {
+			float taunt_chance = 50.0f;
 
+			if (always_succeed) {
+				taunt_chance = 101.0f;
+			} else {
 				if (level_difference < 0) {
-					tauntchance += static_cast<float>(level_difference) * 3.0f;
-					if (tauntchance < 20)
-						tauntchance = 20.0f;
-				}
-
-				else {
-					tauntchance += static_cast<float>(level_difference) * 5.0f;
-					if (tauntchance > 65)
-						tauntchance = 65.0f;
+					taunt_chance += static_cast<float>(level_difference) * 3.0f;
+					if (taunt_chance < 20) {
+						taunt_chance = 20.0f;
+					}
+				} else {
+					taunt_chance += static_cast<float>(level_difference) * 5.0f;
+					if (taunt_chance > 65) {
+						taunt_chance = 65.0f;
+					}
 				}
 			}
 
 			// TauntSkillFalloff rate is not based on any real data. Default of 33% gives a reasonable
 			// result.
-			if (IsClient() && !always_succeed)
-				tauntchance -= (RuleR(Combat, TauntSkillFalloff) *
+			if (IsClient() && !always_succeed) {
+				taunt_chance -= (RuleR(Combat, TauntSkillFalloff) *
 						(CastToClient()->MaxSkill(EQ::skills::SkillTaunt) -
 						 GetSkill(EQ::skills::SkillTaunt)));
+			}
 
-			if (tauntchance < 1)
-				tauntchance = 1.0f;
-
-			tauntchance /= 100.0f;
+			if (taunt_chance < 1) {
+				taunt_chance = 1.0f;
+			}
 
-			success = tauntchance > zone->random.Real(0, 1);
+			taunt_chance /= 100.0f;
+			success = taunt_chance > zone->random.Real(0, 1);
 
 			LogHate(
-				"Taunter mob {} target npc {} tauntchance [{}] success [{}] hate_top [{}]",
+				"Taunter mob {} target npc {} taunt_chance [{}] success [{}] hate_top [{}]",
 				GetMobDescription(),
 				who->GetMobDescription(),
-				tauntchance,
+				taunt_chance,
 				success ? "true" : "false",
 				hate_top ? hate_top->GetMobDescription() : "not found"
 			);
 		}
 
 		if (success) {
 			if (hate_top && hate_top != this) {
-				int64 newhate = (who->GetNPCHate(hate_top) - who->GetNPCHate(this)) + 1 + bonus_hate;
+				int64 new_hate = ((who->GetNPCHate(hate_top) - who->GetNPCHate(this)) + bonus_hate + RuleI(Combat, TauntOverAggro));
```

Comment:
We lost the `+ 1` here, I think that's just so we always have at least `1` hate.

Reply:
Not there, I should have it as the default TauntOverAggro rule no need to have a lingering +1